confgenerator/testdata/valid/linux/metrics-processor_exclude_metrics_type_filter_with_special_chars/input.yaml

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+metrics:
+  processors:
+    metrics_filter:
+      type: exclude_metrics
+      metrics_pattern:
+      # Escaped the comma to work around https://github.com/goccy/go-yaml/issues/269
+      # TODO: remove the escape once the issue is fixed.
+      - agent.googleapis.com/x$y(%28)z\,a':w-+!/*
```

Comment:
this is valid? should we be failing config validation if this doesn't match monitoring api [defined format](http://cloud/monitoring/api/v3/naming-conventions) (plus `.*`)? fine for that to be a separate pr doe

Reply:
Strangely (and sadly), this is a valid metric name. Granted, it's not one that would ever be produced with the `agent.googleapis.com/` prefix, but that's not what this test verifies. The page you are referencing is outdated — see b/207141764.